platform-modules/batch-models/src/main/scala/org/ekstep/analytics/adapter/ContentAdapter.scala

```diff
@@ -30,32 +28,34 @@ object ContentAdapter extends BaseAdapter {
         })
     }
 
-    def getPublishedContent(): Array[Map[String, AnyRef]] = {
+    @tailrec
+    def search(offset: Int, limit: Int, contents: Array[Map[String, AnyRef]], action: (Int, Int) => ContentResult): Array[Map[String, AnyRef]] = {
+        val result = action(offset, limit);
+        val c = contents ++ result.content;
+        if (result.count > (offset + limit)) {
+            search((offset + limit), limit, c, action);
+        } else {
+            c;
+        }
+    }
 
-        @tailrec
-        def search(offset: Int, limit: Int, contents: Array[Map[String, AnyRef]]): Array[Map[String, AnyRef]] = {
-            val result = _search(offset, limit);
-            val c = contents ++ result.content;
-            if (result.count > (offset + limit)) {
-                search((offset + limit), limit, c);
-            } else {
-                c;
-            }
+
+    def getPublishedContent(): Array[Map[String, AnyRef]] = {
+        def _searchContent(offset: Int, limit: Int): ContentResult = {
+            val searchUrl = Constants.getContentSearch();
+            val request = Map("request" -> Map("filters" -> Map("objectType" -> List("Content"), "contentType" -> List("Story", "Worksheet", "Collection", "Game"), "status" -> List("Draft", "Review", "Redraft", "Flagged", "Live", "Retired", "Mock", "Processing", "FlagDraft", "FlagReview")), "exists" -> List("lastPublishedOn", "downloadUrl"), "offset" -> offset, "limit" -> limit));
```

Comment:
Why is the SearchRequest parameters hard-coded? Can this be part of the configuration?

Reply:
this piece of code is not related to the requirement. this change is refactoring of `search` method